### DISCUSSION TITLE
How does the loading skeleton determine state?

### DISCUSSION THREAD

**Discussion by 619** [dvtng/react-loading-skeleton#147] [General]
There doesn't seem to be a prop you can use to manually notify the skeleton loader of the loading state (whether the content is loading or it has finished loading), so I'm wondering how the skeleton loader would be able to deduce when it should be visible.

**Comment by srmagura** [comment]
Thanks for your interest in the library. The `Skeleton` component always represents a loading state (though there is a prop to pause the animation, for advanced use cases.)

So you will only render the skeleton if your data is loading. For example:

```tsx
<div>
  {loading ? <Skeleton /> : data.userName}
</div>
```

Here `data` and `loading` are variables that could come from your data fetching library or a `useState`.
